elements/rh-secondary-nav/rh-secondary-nav-menu.ts

```diff
@@ -0,0 +1,163 @@
+import { html, LitElement } from 'lit';
+import { customElement, property, queryAssignedNodes, query } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+import { pfelement, observed } from '@patternfly/pfe-core/decorators.js';
+import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
+
+import { tabletLandscapeBreakpoint } from '../../lib/tokens.js';
+import { MatchMediaController } from '../../lib/MatchMediaController.js';
+
+import { RhSecondaryNavDropdown } from './rh-secondary-nav-dropdown';
+
+import styles from './rh-secondary-nav-menu.css';
+
+/**
+ * @summary 'A pop up menu for secondary nav, available in full-width and fixed-with sizes'
+ *
+ * @slot base             - Slot to override all shadow dom contents for full-width or fixed-width menus
+ * @slot section          - Hidden if empty, section slot applies an auto column grid system for `<ul>` link lists
+ * @slot links            - Only for type="fixed-width", expects a `<ul>`
+ * @slot cta              - Menu level CTA, expects a `<pfe-cta>`
+ *
+ * @csspart full-width    - {HTMLElement} container - The wrapper for full-width menus
+ * @csspart fixed-width   - {HTMLElement} container - The wrapper for fixed-width menus
+ * @csspart sections      - {HTMLElement} container - The wrapper for menu sections
+ * @csspart cta           - {HTMLElement} container - The wrapper for cta
+ */
+@customElement('rh-secondary-nav-menu') @pfelement()
+export class RhSecondaryNavMenu extends LitElement {
+  static readonly styles = [styles];
+
+  // TODO: change to variant instead of type which is used by html element properties such as input.
+  @property({ reflect: true }) type: 'fixed-width' | 'full-width' = 'full-width';
```

Comment:
in this case what about `layout` or `size`?

Reply:
`layout` is most likely the most appropriate IMHO.  

The `TODO:` comment was a holdover from prior to posting the discussion https://github.com/RedHat-UX/red-hat-design-system/discussions/293 on this very question.  I am open and would like to hear more opinions on this specific instance from the designers and will reference this comment there before making a change.

Reply:
if you ask me, go with `layout` for now, and if we need to change it later per design's input, we'll do that in a subsequent PR

Reply:
Fixed in e441b00b80889be1fa6635a336fbf011e64c5334